pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -406,6 +416,21 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 						// NLBs require a different health check interval than CLBs.
 						// See <https://bugzilla.redhat.com/show_bug.cgi?id=1908758>.
 						service.Annotations[awsLBHealthCheckIntervalAnnotation] = awsLBHealthCheckIntervalNLB
+						var eipAllocationsOfTypeString []string
+						if ci.Spec.EndpointPublishingStrategy != nil {
+							lbSpec := ci.Spec.EndpointPublishingStrategy.LoadBalancer
+							if lbSpec.ProviderParameters.AWS.NetworkLoadBalancerParameters != nil || lbSpec.ProviderParameters.AWS.NetworkLoadBalancerParameters.EIPAllocations != nil || len(lbSpec.ProviderParameters.AWS.NetworkLoadBalancerParameters.EIPAllocations) > 0 {
```

Comment:
Only if `NetworkLoadBalancerParameters != nil` then need to check if  `NetworkLoadBalancerParameters.EIPAllocations != nil` right? 

I see panic might be related here
```
2024-04-30T03:12:04.794Z	INFO	operator.init	runtime/panic.go:914	Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference	{"controller": "ingress_controller", "object": {"name":"default","namespace":"openshift-ingress-operator"}, "namespace": "openshift-ingress-operator", "name": "default", "reconcileID": "4f2dc3b1-abb0-483a-9dd5-e7f21c381a54"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x18f614d]

goroutine 1413 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/ingress-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:116 +0x1e5
panic({0x25e1100?, 0x4609bb0?})
	/usr/lib/golang/src/runtime/panic.go:914 +0x21f
github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress.desiredLoadBalancerService(0xc0014c8600, {{0x2a3cab8, 0x7}, {0x2a42282, 0xa}, {0xc000ae7fd0, 0xe}, {0xc0017fe060, 0x24}, 0xc0011107ea, ...}, ...)
	/ingress-operator/pkg/operator/controller/ingress/load_balancer_service.go:422 +0x9ad

```

Reply:
@lihongan I don't see these nil errors. Which test are you seeing this ?

Reply:
just set `spec.endpointPublishingStrategy` as below (note: do not set `nlbParameters: {}`)
```
spec:
  endpointPublishingStrategy:
    loadBalancer:
      dnsManagementPolicy: Managed
      providerParameters:
        aws:
          type: NLB
        type: AWS
      scope: External
    type: LoadBalancerService
```

Reply:
This seems be an API related issue. I do have copied all the crds in the manifest dir. The CVO should pick the right one based on the `release.openshift.io/feature-set` annotation

After spinning the cluster from cluster-bot:

Run:
git clone my branch

oc scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator

oc replace -f manifests/0000_50_ingress_00_ingresscontrollers-TechPreviewNoUpgrade.crd.yaml

I use the above commands in my cluster and have not seen this issue.